diff --git a/arch/arm/boot/dts/zynq-coraz7s.dtsi b/arch/arm/boot/dts/zynq-coraz7s.dtsi
index b079648d2b07..4fd4d82b50ab 100644
--- a/arch/arm/boot/dts/zynq-coraz7s.dtsi
+++ b/arch/arm/boot/dts/zynq-coraz7s.dtsi
@@ -28,6 +28,14 @@ chosen {
 		stdout-path = "serial0:115200n8";
 	};
 
+	usb_phy0: phy0@e0002000 {
+		compatible = "ulpi-phy";
+		#phy-cells = <0>;
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
+
 	fpga_axi: fpga-axi@0 {
 		compatible = "simple-bus";
 		#address-cells = <0x1>;
@@ -73,5 +81,5 @@ &uart0 {
 &usb0 {
 	status = "okay";
 	dr_mode = "host";
+	usb-phy = <&usb_phy0>;
 };
-
diff --git a/arch/arm/boot/dts/zynq-m2k.dtsi b/arch/arm/boot/dts/zynq-m2k.dtsi
index b82bf5a25641..6a80b110a07a 100644
--- a/arch/arm/boot/dts/zynq-m2k.dtsi
+++ b/arch/arm/boot/dts/zynq-m2k.dtsi
@@ -19,6 +19,15 @@ memory {
 	chosen {
 		stdout-path = "/amba@0/uart@E0001000";
 	};
+
+	usb_phy0: phy0 {
+		compatible = "ulpi-phy";
+		#phy-cells = <0>;
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
+
 };
 
 &cpu0 {
@@ -46,6 +55,7 @@ &usb0 {
 	xlnx,phy-reset-gpio = <&gpio0 52 0>;
 	dr_mode = "otg";
 	status = "okay";
+	usb-phy = <&usb_phy0>;
 };
 
 &clkc {
@@ -94,5 +104,3 @@ channel@0 {
 		};
 	};
 };
-
-
diff --git a/arch/arm/boot/dts/zynq-pluto-sdr-revc.dts b/arch/arm/boot/dts/zynq-pluto-sdr-revc.dts
index fe5569225fd3..1483505ae12f 100644
--- a/arch/arm/boot/dts/zynq-pluto-sdr-revc.dts
+++ b/arch/arm/boot/dts/zynq-pluto-sdr-revc.dts
@@ -98,6 +98,11 @@ &cf_ad9364_dac_core_0 {
 / {
 	model = "Analog Devices PlutoSDR Rev.C (Z7010/AD9363)";
 
+	aliases {
+		ethernet0 = &gem0;
+		mmc0 = &sdhci0;
+	};
+
 	leds {
 		compatible = "gpio-leds";
 		led0 {
@@ -140,3 +145,18 @@ channel@1 {
 		};
 	};
 };
+
+&gem0 {
+	status = "okay";
+
+    pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gem0_default>;
+};
+
+&usb0 {
+	status = "okay";
+	dr_mode = "otg";
+	usb-phy = <&usb_phy0>;
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_usb0_default>;
+};
diff --git a/arch/arm/boot/dts/zynq-pluto-sdr.dtsi b/arch/arm/boot/dts/zynq-pluto-sdr.dtsi
index 08c5a42204f3..0fe5bba8a082 100644
--- a/arch/arm/boot/dts/zynq-pluto-sdr.dtsi
+++ b/arch/arm/boot/dts/zynq-pluto-sdr.dtsi
@@ -30,10 +30,26 @@ ad9364_clkin: clock@0 {
 		};
 	};
 
+	usb_phy0: phy0 {
+		compatible = "ulpi-phy";
+		#phy-cells = <0>;
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
+
+};
+
+&uart1 {
+	u-boot,dm-pre-reloc;
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1_default>;
 };
 
 &sdhci0 {
-	status = "disabled";
+	status = "okay";
+    disable-wp;
 };
 
 &watchdog0 {
@@ -42,9 +58,24 @@ &watchdog0 {
 };
 
 &usb0 {
-	xlnx,phy-reset-gpio = <&gpio0 52 0>;
+	xlnx,phy-reset-gpio = <&gpio0 46 0>;
 	dr_mode = "otg";
 	status = "okay";
+	usb-phy = <&usb_phy0>;
+};
+
+&gem0 {
+	status = "ok";
+	phy-mode = "rgmii-id";
+	phy-handle = <&ethernet_phy>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gem0_default>;
+
+
+	ethernet_phy: ethernet-phy@1 {
+		reg = <1>;
+		device_type = "ethernet-phy";
+	};
 };
 
 &qspi {
@@ -90,6 +121,18 @@ channel@0 {
 	};
 };
 
+
+&gem0 {
+	status = "disabled";
+	phy-mode = "rgmii-id";
+	phy-handle = <&ethernet_phy>;
+
+	ethernet_phy: ethernet-phy@1 {
+		reg = <1>;
+		device_type = "ethernet-phy";
+	};
+};
+
 / {
 	fpga_axi: fpga-axi@0 {
 		compatible = "simple-bus";
@@ -179,6 +222,148 @@ mwipcore@43c00000 {
 	};
 };
 
+&pinctrl0 {
+	pinctrl_gem0_default: gem0-default {
+			mux {
+				function = "ethernet0";
+				groups = "ethernet0_0_grp";
+			};
+
+			conf {
+				groups = "ethernet0_0_grp";
+				slew-rate = <0>;
+				io-standard = <1>;
+			};
+
+			conf-rx {
+				pins = "MIO22", "MIO23", "MIO24", "MIO25", "MIO26", "MIO27";
+				bias-high-impedance;
+				low-power-disable;
+			};
+
+			conf-tx {
+				pins = "MIO16", "MIO17", "MIO18", "MIO19", "MIO20", "MIO21";
+				low-power-enable;
+				bias-disable;
+			};
+
+			mux-mdio {
+				function = "mdio0";
+				groups = "mdio0_0_grp";
+			};
+
+			conf-mdio {
+				groups = "mdio0_0_grp";
+				slew-rate = <0>;
+				io-standard = <1>;
+				bias-disable;
+			};
+	};
+
+	pinctrl_usb0_default: usb0-default {
+		mux {
+			groups = "usb0_0_grp";
+			function = "usb0";
+		};
+
+		conf {
+			groups = "usb0_0_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+
+		conf-rx {
+			pins = "MIO29", "MIO31", "MIO36";
+			bias-high-impedance;
+		};
+
+		conf-pull-up {
+			pins = "MIO46";
+			bias-pull-up;
+		};
+
+		conf-tx {
+			pins = "MIO28", "MIO30", "MIO32", "MIO33", "MIO34",
+					"MIO35", "MIO37", "MIO38", "MIO39";
+			bias-disable;
+		};
+	};
+
+	pinctrl_gpio0_default: gpio0-default {
+		mux {
+			function = "gpio0";
+			groups = "gpio0_7_grp", "gpio0_11_grp", "gpio0_14_grp", "gpio0_46_grp";
+		};
+
+		conf {
+			groups = "gpio0_7_grp", "gpio0_11_grp", "gpio0_14_grp", "gpio0_46_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+
+		conf-pull-up {
+			pins = "MIO14";
+			bias-pull-up;
+		};
+
+		conf-pull-none {
+			pins = "MIO7", "MIO11";
+			bias-disable;
+		};
+	};
+
+	pinctrl_sdhci0_default: sdhci0-default {
+		mux {
+			groups = "sdio0_2_grp";
+			function = "sdio0";
+		};
+
+		conf {
+			groups = "sdio0_2_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+			bias-disable;
+		};
+
+		mux-cd {
+			groups = "gpio0_47_grp";
+			function = "sdio0_cd";
+		};
+
+		conf-cd {
+			groups = "gpio0_47_grp";
+			bias-high-impedance;
+			bias-pull-up;
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+	};
+
+	pinctrl_uart1_default: uart1-default {
+		mux {
+			groups = "uart1_1_grp";
+			function = "uart1";
+		};
+
+		conf {
+			groups = "uart1_1_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+
+		conf-rx {
+			pins = "MIO13";
+			bias-high-impedance;
+		};
+
+		conf-tx {
+			pins = "MIO12";
+			bias-disable;
+		};
+	};
+};
+
+
 &spi0 {
 	status = "okay";
 
diff --git a/arch/arm/boot/dts/zynq-zc702.dts b/arch/arm/boot/dts/zynq-zc702.dts
index aa084a9938d9..b91785c93349 100644
--- a/arch/arm/boot/dts/zynq-zc702.dts
+++ b/arch/arm/boot/dts/zynq-zc702.dts
@@ -57,6 +57,14 @@ led-ds23 {
 			linux,default-trigger = "heartbeat";
 		};
 	};
+
+	usb_phy0: phy0@e0002000 {
+		compatible = "ulpi-phy";
+		#phy-cells = <0>;
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
 };
 
 &can0 {
@@ -441,6 +449,7 @@ &uart1 {
 &usb0 {
 	status = "okay";
 	dr_mode = "host";
+    usb-phy = <&usb_phy0>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usb0_default>;
 };
diff --git a/arch/arm/boot/dts/zynq-zc706.dts b/arch/arm/boot/dts/zynq-zc706.dts
index 6a6e8312b3f7..cb6e9117f3a3 100644
--- a/arch/arm/boot/dts/zynq-zc706.dts
+++ b/arch/arm/boot/dts/zynq-zc706.dts
@@ -27,6 +27,14 @@ chosen {
 		bootargs = "";
 		stdout-path = "serial0:115200n8";
 	};
+
+	usb_phy0: phy0@e0002000 {
+		compatible = "ulpi-phy";
+		#phy-cells = <0>;
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
 };
 
 &clkc {
@@ -352,6 +360,7 @@ &uart1 {
 &usb0 {
 	status = "okay";
 	dr_mode = "host";
+	usb-phy = <&usb_phy0>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usb0_default>;
 };
diff --git a/arch/arm/boot/dts/zynq-zed.dts b/arch/arm/boot/dts/zynq-zed.dts
index 60f73cd2601b..2d531a6ea2ca 100644
--- a/arch/arm/boot/dts/zynq-zed.dts
+++ b/arch/arm/boot/dts/zynq-zed.dts
@@ -26,6 +26,14 @@ chosen {
 		bootargs = "";
 		stdout-path = "serial0:115200n8";
 	};
+
+	usb_phy0: phy0@e0002000 {
+		compatible = "ulpi-phy";
+		#phy-cells = <0>;
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
+	};
 };
 
 &clkc {
@@ -93,4 +101,5 @@ &uart1 {
 &usb0 {
 	status = "okay";
 	dr_mode = "host";
+	usb-phy = <&usb_phy0>;
 };
diff --git a/arch/arm/configs/zynq_pluto_defconfig b/arch/arm/configs/zynq_pluto_defconfig
index 506eec34f4b4..818b84bddc05 100644
--- a/arch/arm/configs/zynq_pluto_defconfig
+++ b/arch/arm/configs/zynq_pluto_defconfig
@@ -269,3 +269,15 @@ CONFIG_RCU_CPU_STALL_TIMEOUT=60
 CONFIG_DEBUG_LL=y
 CONFIG_DEBUG_ZYNQ_UART1=y
 CONFIG_EARLY_PRINTK=y
+
+CONFIG_OF_MDIO=y
+CONFIG_REALTEK_PHY=y
+CONFIG_ETHERNET=y
+CONFIG_NET_VENDOR_XILINX=y
+CONFIG_XILINX_AXI_EMAC=y
+CONFIG_MACB=y
+
+CONFIG_MMC=y
+CONFIG_MMC_SDHCI=y
+CONFIG_MMC_SDHCI_PLTFM=y
+CONFIG_MMC_SDHCI_OF_ARASAN=y
