diff --git a/arch/arm/dts/zynq-pluto-sdr.dts b/arch/arm/dts/zynq-pluto-sdr.dts
index c618d98b54..c318383554 100644
--- a/arch/arm/dts/zynq-pluto-sdr.dts
+++ b/arch/arm/dts/zynq-pluto-sdr.dts
@@ -16,6 +16,7 @@
 		ethernet0 = &gem0;
 		serial0 = &uart1;
 		spi0 = &qspi;
+		mmc0 = &sdhci0;
 	};
 
 	memory {
@@ -75,3 +76,8 @@
 	dr_mode = "host";
 	usb-phy = <&usb_phy0>;
 };
+
+&sdhci0 {
+	u-boot,dm-pre-reloc;
+	status = "okay";
+};
\ No newline at end of file
diff --git a/configs/zynq_pluto_defconfig b/configs/zynq_pluto_defconfig
index 0daf46221e..5991fa6cf7 100644
--- a/configs/zynq_pluto_defconfig
+++ b/configs/zynq_pluto_defconfig
@@ -7,13 +7,14 @@ CONFIG_SPL=y
 CONFIG_FIT=y
 CONFIG_FIT_VERBOSE=y
 CONFIG_FIT_SIGNATURE=y
-CONFIG_BOOTDELAY=0
+CONFIG_BOOTDELAY=3
 CONFIG_SYS_NO_FLASH=y
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="Pluto> "
 # CONFIG_CMD_IMLS is not set
 # CONFIG_CMD_XIMG is not set
 # CONFIG_CMD_FLASH is not set
+CONFIG_CMD_MMC=y
 CONFIG_CMD_SF=y
 CONFIG_CMD_SPI=y
 CONFIG_CMD_USB=y
@@ -54,3 +55,11 @@ CONFIG_G_DNL_VENDOR_NUM=0x0456
 CONFIG_G_DNL_PRODUCT_NUM=0xb674
 CONFIG_REGEX=y
 CONFIG_LIB_RAND=y
+
+CONFIG_NET_RANDOM_ETHADDR=y
+
+CONFIG_DM_MMC=y
+CONFIG_ZYNQ_SDHCI=y
+CONFIG_NAND_ARASAN=y
+CONFIG_DM_ETH=y
+CONFIG_ZYNQ_GEM=y
diff --git a/include/configs/zynq-common.h b/include/configs/zynq-common.h
index fd50caf450..9765ca4a08 100644
--- a/include/configs/zynq-common.h
+++ b/include/configs/zynq-common.h
@@ -257,7 +257,7 @@
 	"fdt_high=0x20000000\0"	\
 	"initrd_high=0x20000000\0"	\
 	"bootenv=uEnv.txt\0" \
-	"maxcpus=1\0" \
+	"maxcpus=2\0" \
 	"clear_reset_cause=mw f8000008 df0d && mw f8000258 00400000 && mw f8000004 767b\0" \
 	"loadbootenv=load mmc 0 ${loadbootenv_addr} ${bootenv}\0" \
 	"importbootenv=echo Importing environment from SD ...; " \
@@ -269,7 +269,7 @@
 			"fi; " \
 		"fi; \0" \
 	"refclk_source=internal\0" \
-	"mode=1r1t\0" \
+	"mode=2r2t\0" \
 	"adi_loadvals_pluto=if test -n \"${ad936x_ext_refclk}\" && test ! -n \"${ad936x_skip_ext_refclk}\"; then " \
 			"fdt set /clocks/clock@0 clock-frequency ${ad936x_ext_refclk}; " \
 		"fi; " \
