diff --git a/arch/arm/dts/zynq-pluto-sdr.dts b/arch/arm/dts/zynq-pluto-sdr.dts
index c618d98b54..684826ebe6 100644
--- a/arch/arm/dts/zynq-pluto-sdr.dts
+++ b/arch/arm/dts/zynq-pluto-sdr.dts
@@ -16,6 +16,7 @@
 		ethernet0 = &gem0;
 		serial0 = &uart1;
 		spi0 = &qspi;
+		mmc0 = &sdhci0;
 	};
 
 	memory {
@@ -66,12 +67,167 @@
 	};
 };
 
+&gpio0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_gpio0_default>;
+};
+
 &uart1 {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1_default>;
 };
 
 &usb0 {
 	status = "okay";
 	dr_mode = "host";
 	usb-phy = <&usb_phy0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usb0_default>;
+};
+
+&sdhci0 {
+	u-boot,dm-pre-reloc;
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_sdhci0_default>;
 };
+
+&pinctrl0 {
+	pinctrl_gem0_default: gem0-default {
+			mux {
+				function = "ethernet0";
+				groups = "ethernet0_0_grp";
+			};
+
+			conf {
+				groups = "ethernet0_0_grp";
+				slew-rate = <0>;
+				io-standard = <1>;
+			};
+
+			conf-rx {
+				pins = "MIO22", "MIO23", "MIO24", "MIO25", "MIO26", "MIO27";
+				bias-high-impedance;
+				low-power-disable;
+			};
+
+			conf-tx {
+				pins = "MIO16", "MIO17", "MIO18", "MIO19", "MIO20", "MIO21";
+				low-power-enable;
+				bias-disable;
+			};
+
+			mux-mdio {
+				function = "mdio0";
+				groups = "mdio0_0_grp";
+			};
+
+			conf-mdio {
+				groups = "mdio0_0_grp";
+				slew-rate = <0>;
+				io-standard = <1>;
+				bias-disable;
+			};
+	};
+
+	pinctrl_usb0_default: usb0-default {
+		mux {
+			groups = "usb0_0_grp";
+			function = "usb0";
+		};
+
+		conf {
+			groups = "usb0_0_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+
+		conf-rx {
+			pins = "MIO29", "MIO31", "MIO36";
+			bias-high-impedance;
+			slew-rate = <0>;
+			io-standard = <1>;
+			bias-disable;
+		};
+
+		conf-tx {
+			pins = "MIO28", "MIO30", "MIO32", "MIO33", "MIO34",
+					"MIO35", "MIO37", "MIO38", "MIO39";
+			bias-disable;
+		};
+	};
+
+	pinctrl_gpio0_default: gpio0-default {
+		mux {
+			function = "gpio0";
+			groups = "gpio0_7_grp", "gpio0_11_grp", "gpio0_14_grp", "gpio0_46_grp";
+		};
+
+		conf {
+			groups = "gpio0_7_grp", "gpio0_11_grp", "gpio0_14_grp", "gpio0_46_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+
+		conf-pull-up {
+			pins = "MIO14";
+			bias-pull-up;
+		};
+
+		conf-pull-none {
+			pins = "MIO7", "MIO11", "MIO46";
+			bias-disable;
+		};
+	};
+
+	pinctrl_sdhci0_default: sdhci0-default {
+		mux {
+			groups = "sdio0_2_grp";
+			function = "sdio0";
+		};
+
+		conf {
+			groups = "sdio0_2_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+			bias-disable;
+		};
+
+		mux-cd {
+			groups = "gpio0_47_grp";
+			function = "sdio0_cd";
+		};
+
+		conf-cd {
+			groups = "gpio0_47_grp";
+			bias-high-impedance;
+			bias-pull-up;
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+	};
+
+	pinctrl_uart1_default: uart1-default {
+		mux {
+			groups = "uart1_1_grp";
+			function = "uart1";
+		};
+
+		conf {
+			groups = "uart1_1_grp";
+			slew-rate = <0>;
+			io-standard = <1>;
+		};
+
+		conf-rx {
+			pins = "MIO13";
+			bias-high-impedance;
+		};
+
+		conf-tx {
+			pins = "MIO12";
+			bias-disable;
+		};
+	};
+};
\ No newline at end of file
diff --git a/configs/zynq_pluto_defconfig b/configs/zynq_pluto_defconfig
index 0daf46221e..9006b8bcf4 100644
--- a/configs/zynq_pluto_defconfig
+++ b/configs/zynq_pluto_defconfig
@@ -1,5 +1,5 @@
 CONFIG_ARM=y
-CONFIG_SYS_CONFIG_NAME="zynq_zc70x"
+CONFIG_SYS_CONFIG_NAME="zynq_pluto"
 CONFIG_ARCH_ZYNQ=y
 CONFIG_SYS_MALLOC_F_LEN=0x800
 CONFIG_DEFAULT_DEVICE_TREE="zynq-pluto-sdr"
@@ -7,13 +7,14 @@ CONFIG_SPL=y
 CONFIG_FIT=y
 CONFIG_FIT_VERBOSE=y
 CONFIG_FIT_SIGNATURE=y
-CONFIG_BOOTDELAY=0
+CONFIG_BOOTDELAY=3
 CONFIG_SYS_NO_FLASH=y
 CONFIG_HUSH_PARSER=y
 CONFIG_SYS_PROMPT="Pluto> "
 # CONFIG_CMD_IMLS is not set
 # CONFIG_CMD_XIMG is not set
 # CONFIG_CMD_FLASH is not set
+CONFIG_CMD_MMC=y
 CONFIG_CMD_SF=y
 CONFIG_CMD_SPI=y
 CONFIG_CMD_USB=y
@@ -54,3 +55,11 @@ CONFIG_G_DNL_VENDOR_NUM=0x0456
 CONFIG_G_DNL_PRODUCT_NUM=0xb674
 CONFIG_REGEX=y
 CONFIG_LIB_RAND=y
+
+CONFIG_NET_RANDOM_ETHADDR=y
+
+CONFIG_DM_MMC=y
+CONFIG_ZYNQ_SDHCI=y
+CONFIG_NAND_ARASAN=y
+CONFIG_DM_ETH=y
+CONFIG_ZYNQ_GEM=y
\ No newline at end of file
diff --git a/include/configs/zynq-common.h b/include/configs/zynq-common.h
index 50e93061a9..f7c2e010b8 100644
--- a/include/configs/zynq-common.h
+++ b/include/configs/zynq-common.h
@@ -257,7 +257,7 @@
 	"fdt_high=0x20000000\0"	\
 	"initrd_high=0x20000000\0"	\
 	"bootenv=uEnv.txt\0" \
-	"maxcpus=1\0" \
+	"maxcpus=2\0" \
 	"clear_reset_cause=mw f8000008 df0d && mw f8000258 00400000 && mw f8000004 767b\0" \
 	"loadbootenv=load mmc 0 ${loadbootenv_addr} ${bootenv}\0" \
 	"importbootenv=echo Importing environment from SD ...; " \
diff --git a/include/configs/zynq_pluto.h b/include/configs/zynq_pluto.h
new file mode 100644
index 0000000000..8c0e74d770
--- /dev/null
+++ b/include/configs/zynq_pluto.h
@@ -0,0 +1,24 @@
+/*
+ * (C) Copyright 2013 Xilinx, Inc.
+ *
+ * Configuration settings for the PlutoSDR Boards
+ * See zynq-common.h for Zynq common configs
+ *
+ * SPDX-License-Identifier:	GPL-2.0+
+ */
+
+#ifndef __CONFIG_ZYNQ_PLUTOSDR_H
+#define __CONFIG_ZYNQ_PLUTOSDR_H
+
+#define CONFIG_ZYNQ_I2C0
+#define CONFIG_ZYNQ_EEPROM
+#define CONFIG_ZYNQ_USB
+#define CONFIG_DFU_SF
+
+#define CONFIG_MII
+#define CONFIG_PHY_ADDR 0
+#define CONFIG_HAS_ETH0
+
+#include <configs/zynq-common.h>
+
+#endif /* __CONFIG_ZYNQ_PLUTOSDR_H */
